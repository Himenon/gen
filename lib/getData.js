const fs = require('fs')
const path = require('path')
const { promisify } = require('util')
const loadJSON = require('load-json-file')
const matter = require('gray-matter')

const readdir = promisify(fs.readdir)
const readFile = promisify(fs.readFile)

const getContent = async dirname => {
  const theme = await loadJSON(path.join(dirname, 'theme.json'))
  const lab = await loadJSON(path.join(dirname, 'lab.json'))
  const filenames = await readdir(dirname)
  const jsxFilenames = filenames.filter(name => /\.jsx$/.test(name))
  const mdFilenames = filenames.filter(name => /\.md/.test(name))

  const contentFiles = [ ...jsxFilenames, ...mdFilenames ]
  const promises = contentFiles.map(getPage(dirname))
  const pages = await Promise.all(promises)
  const withLayouts = pages.map(getLayout(pages))

  return {
    dirname,
    theme,
    lab,
    pages: withLayouts
  }
}

const getPage = dirname => async filename => {
  const ext = path.extname(filename)
  const name = path.basename(filename, ext)
  const raw = await readFile(path.join(dirname, filename), 'utf8')
  const { data, content } = matter(raw)

  return {
    filename,
    name,
    ext,
    raw,
    data,
    content,
  }
}

const getLayout = pages => page => {
  if (page.ext !== '.md') return page
  if (!page.data.layout) return page
  const layout = pages.find(p => p.name === page.data.layout)
  if (!layout) return page
  page.data = Object.assign({}, layout.data, page.data)
  page.layoutJSX = layout.content
  return page
}

module.exports = getContent
