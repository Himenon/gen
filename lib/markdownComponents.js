const createComponents = require('./createComponents')

module.exports = opts => createComponents([
  {
    name: 'Title',
    type: 'h1',
    style: {},
    props: {
      mt: 4,
      mb: 2
    }
  },
  {
    name: 'Heading',
    type: 'h2',
    style: {},
    props: {
      mt: 4,
      mb: 2,
    }
  },
  {
    name: 'Subhead',
    type: 'h3',
    style: {},
    props: {
      mt: 3,
      mb: 2
    }
  },
  /*
  { name: 'Text', },
  { name: 'Link', },
  { name: 'Pre', },
  { name: 'Code', },
  { name: 'Table', },
  */
], opts)
