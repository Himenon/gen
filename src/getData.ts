import * as fs from 'fs'
import * as path from 'path'
import { promisify } from 'util'
import * as loadJSON from 'load-json-file'
import * as matter from 'gray-matter'

const readdir = promisify(fs.readdir)

export const getContent = async (dirname: string, opts: any) => {
  let theme = {}
  let lab = { components: [] }
  try {
    theme = await loadJSON(path.join(dirname, 'theme.json'))
  } catch (err) {
    // console.log('no theme.json found')
  }

  try {
    lab = await loadJSON(path.join(dirname, 'lab.json'))
  } catch (err) {
    // console.log('no lab.json found')
  }

  const allFiles = await readdir(dirname)
  const filenames = allFiles.filter(name => !/^\./.test(name))
  const jsxFilenames = filenames.filter(name => /\.jsx$/.test(name))
  const mdFilenames = filenames.filter(name => /\.md/.test(name))

  const contentFiles = [...jsxFilenames, ...mdFilenames]
  const promises = contentFiles.map(getPage(dirname))
  const pages = await Promise.all(promises)
  const withLayouts = pages.map(getLayout(pages))

  return {
    dirname,
    theme,
    lab,
    pages: withLayouts,
  }
}

const getPage = (dirname: string) => async (filename: string) => {
  const ext = path.extname(filename)
  const name = path.basename(filename, ext)
  // const raw = await readFile(path.join(dirname, filename), 'utf8')
  const raw = fs.readFileSync(path.join(dirname, filename), 'utf8')
  const { data, content } = matter(raw)

  return {
    filename,
    name,
    ext,
    raw,
    data,
    content,
  }
}

const getLayout = (pages: any) => (page: any) => {
  if (page.ext !== '.md') return page
  if (!page.data.layout) return page
  const layout = pages.find((p: any) => p.name === page.data.layout)
  if (!layout) return page
  page.data = Object.assign({}, layout.data, page.data)
  page.layoutJSX = layout.content
  return page
}

export { getContent as getData }
