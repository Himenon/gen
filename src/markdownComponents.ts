import { createComponents } from './createComponents'
import { Options2 } from './types'

const gray = 'rgba(0, 0, 0, .125)'

export const markdownComponents = (opts: Options2) =>
  createComponents(
    [
      {
        name: 'Title',
        type: 'h1',
        style: {},
        props: {
          mt: 4,
          mb: 2,
        },
        system: [],
      },
      {
        name: 'Heading',
        type: 'h2',
        style: {},
        props: {
          mt: 4,
          mb: 2,
        },
        system: [],
      },
      {
        name: 'Subhead',
        type: 'h3',
        style: {},
        props: {
          mt: 4,
          mb: 2,
        },
        system: [],
      },
      {
        name: 'Pre',
        type: 'pre',
        style: {
          fontFamily: 'Menlo, monospace',
          overflow: 'auto',
        },
        props: {
          fontSize: 14,
          p: 0,
          pl: 3,
          mt: 3,
          mb: 3,
          borderWidth: 4,
          borderLeft: true,
          borderColor: gray,
        },
        system: ['borderWidth', 'borderColor'],
      },
      {
        name: 'Code',
        type: 'code',
        style: {
          fontFamily: 'Menlo, monospace',
        },
        props: {
          fontSize: 14,
        },
      },
      {
        name: 'Table',
        type: 'table',
        style: {
          borderCollapse: 'separate',
          borderSpacing: 0,
          maxWidth: '100%',
          width: '100%',
          '& th': {
            textAlign: 'left',
            fontWeight: 'bold',
            lineHeight: 'inherit',
            verticalAlign: 'bottom',
            padding: '4px',
            borderBottomWidth: '2px',
            borderBottomStyle: 'solid',
            borderBottomColor: gray,
          },
          '& td': {
            verticalAlign: 'top',
            lineHeight: 'inherit',
            padding: '4px',
            borderBottomWidth: '1px',
            borderBottomStyle: 'solid',
            borderBottomColor: gray,
          },
        },
        props: {
          mt: 3,
          mb: 3,
        },
        system: [],
      },
      {
        name: 'Divider',
        type: 'hr',
        style: {
          border: 0,
        },
        props: {
          my: 3,
          borderWidth: 1,
          borderBottom: true,
          borderColor: gray,
        },
        system: ['borderWidth', 'borderColor'],
      },
      {
        name: 'Blockquote',
        type: 'blockquote',
        style: {},
        props: {
          px: 3,
          mx: 0,
          my: 3,
          fontWeight: 'bold',
          borderWidth: 4,
          borderLeft: true,
          borderColor: gray,
        },
        system: ['fontWeight', 'borderWidth', 'borderColor'],
      },
    ],
    opts,
  )
