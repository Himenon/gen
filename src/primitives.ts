import { ComponentConfig } from '@gen'
import { space, themeGet, width } from 'styled-system'

export const primitives: ComponentConfig[] = [
  {
    name: 'Box',
    type: 'div',
    props: {},
    style: {},
    system: [],
  },
  {
    name: 'Flex',
    type: 'Box',
    props: {},
    style: {
      display: 'flex',
    },
    system: ['alignItems', 'justifyContent', 'flexWrap', 'flexDirection'],
  },
  {
    name: 'Grid',
    type: 'div',
    props: {},
    style: (props: ComponentConfig) => ({
      display: 'flex',
      flexWrap: 'wrap',
      // disable defaults from dxs
      width: '100% !important',
      margin: '0 !important',
      padding: '0 !important',
      color: 'inherit !important',
      backgroundColor: 'transparent !important',
      '& > *': { ...space(props), ...width(props) },
    }),
    system: ['alignItems', 'justifyContent', 'flexDirection'],
  },
  {
    name: 'Text',
    type: 'div',
    props: {},
    style: {},
    system: ['textAlign', 'fontWeight'],
  },
  {
    name: 'Heading',
    type: 'h2',
    props: {
      m: 0,
    },
    style: {
      lineHeight: 1.25,
    },
    system: ['textAlign', 'fontWeight'],
  },
  {
    name: 'Link',
    type: 'a',
    props: {
      color: 'blue',
    },
    style: {},
    system: [],
  },
  {
    name: 'Image',
    type: 'img',
    props: {},
    style: {
      maxWidth: '100%',
      height: 'auto',
    },
    system: [],
  },
  {
    name: 'Font',
    type: 'div',
    props: {},
    style: (props: ComponentConfig) => ({
      fontFamily: themeGet('fonts.0', 'sans-serif')(props),
      lineHeight: themeGet('lineHeight', '1.5')(props),
      color: themeGet('textColor')(props),
    }),
    system: [],
  },
  {
    name: 'DefaultLayout',
    type: 'div',
    props: {
      mx: 'auto',
      px: 3,
      pt: 3,
      pb: 4,
    },
    style: {
      maxWidth: '768px',
    },
    system: [],
  },
]
