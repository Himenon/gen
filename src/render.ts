import * as webfont from '@compositor/webfont'
import * as dot from 'dot-prop'
import * as glamor from 'glamor/server'
import * as glamorous from 'glamorous'
import * as React from 'react'
import { renderToStaticMarkup, renderToString } from 'react-dom/server'
import * as SC from 'styled-components'

import { createComponents } from './createComponents'
import { createHTML } from './createHTML'
import { toComponent } from './jsx'
import Markdown from './Markdown'
import primitives from './primitives'

import { Content, CreateHtmlData, FirstPage, Options, Options2, ScopedComponents } from './types'

export type BasicComponentProps = CreateHtmlData & React.Attributes

export interface RenderPage extends FirstPage {
  html: string
}

const h = React.createElement

const themeProviders = {
  'styled-components': SC.ThemeProvider,
  glamorous: glamorous.ThemeProvider,
}

type CssCreator = (Component: React.ComponentClass, props: BasicComponentProps) => string

const cssCreators: { [key: string]: CssCreator } = {
  'styled-components': (Component, props) => {
    const sheet = new SC.ServerStyleSheet()
    renderToStaticMarkup(sheet.collectStyles(h(Component, props)))
    const tags = sheet.getStyleTags()
    return tags
  },
  glamorous: (Component, props) => {
    const { css } = glamor.renderStatic(() => renderToString(h(Component, props)))
    const tag = `<style>${css}</style>`
    return tag
  },
}

const getLayout = (pages: FirstPage[] = [], data: CreateHtmlData, scope: ScopedComponents) => {
  if (!data.layout) {
    return scope.DefaultLayout
  }
  const layout: FirstPage | undefined = pages.find(page => page.name === data.layout)

  if (!layout || layout.ext !== '.jsx') {
    return scope.DefaultLayout
  }

  const { content } = layout
  try {
    const Comp = toComponent(content, scope)
    Comp.defaultProps = data
    return Comp
  } catch (err) {
    console.log(err)
    return scope.DefaultLayout
  }
}

// theme,
const renderPage = (scope: ScopedComponents, opts: Options2) => (page: FirstPage): RenderPage => {
  const library = opts.library
  const Provider = (library && themeProviders[library]) || themeProviders['styled-components']
  const getCSS: CssCreator = (library && cssCreators[library]) || cssCreators['styled-components']

  const Layout = page.ext === '.md' ? getLayout(opts.pages, page.data, scope) : React.Fragment
  const pageScope = {
    ...scope,
    Layout,
    Markdown,
    Provider,
    scope,
    page,
    library,
    options: opts,
  }

  const content =
    page.ext === '.jsx'
      ? page.content
      : `<Markdown
        text={page.content}
        scope={scope}
        library='${library}'
        options={options}
      />`

  const Page = toComponent(
    `<Provider theme={theme}>
      <Font>
        <Layout>
          ${content}
        </Layout>
      </Font>
    </Provider>`,
    pageScope,
  )

  try {
    const el = h(Page, page.data)
    const body = renderToStaticMarkup(el)
    // const html = renderToString(el)
    // todo: css
    const css = getCSS(Page, page.data) // full style tag
    const fontLinks = dot
      .get(scope, 'theme.fonts', [])
      .map((font: string) => webfont.getLinkTag(font, []))
      .filter((tag: string | false) => !!tag)
      .join('')
    const html = createHTML({ body, css, fontLinks, data: page.data })
    return {
      ...page,
      html,
    }
  } catch (err) {
    console.log(err)
    return {
      ...page,
      html: err.toString(),
    }
  }
}

const render = async ({ dirname, theme = {}, lab = {}, pages = [] }: Content, opts: Options): Promise<RenderPage[]> => {
  const library = lab.library || 'styled-components'
  const opts2: Options2 = {
    dirname,
    library,
    pages,
    ...opts,
  }

  const base = createComponents(primitives, opts2)
  const components = createComponents(lab.components || [], opts2)
  const scope: ScopedComponents = {
    ...base,
    ...components,
    theme,
  }
  const rendered = pages.map(renderPage(scope, opts2))

  return rendered
}

export { render }
