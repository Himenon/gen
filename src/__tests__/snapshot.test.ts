import * as path from 'path'
import { getData } from '../getData'
import { markdownComponents } from '../markdownComponents'
import { primitives } from '../primitives'
import { render } from '../render'

const dirname = path.join(__dirname, '../../examples')

describe('snapshot', () => {
  test('getData reads theme.json, lab.json, .jsx, and .md files', async () => {
    const data = await getData(dirname, {})
    expect(data).toMatchSnapshot()
  })

  test('render creates pages', async () => {
    const data = await getData(dirname, {})
    const pages = await render(data, {})
    expect(pages).toMatchSnapshot()
  })

  test('exports primitives config', () => {
    expect(primitives).toMatchSnapshot()
  })

  test('generates markdown components', () => {
    const components = markdownComponents({})
    expect(components).toMatchSnapshot()
  })
})
