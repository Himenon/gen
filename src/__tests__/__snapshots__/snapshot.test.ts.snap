// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot exports primitives config 1`] = `
Array [
  Object {
    "name": "Box",
    "props": Object {},
    "style": Object {},
    "system": Array [],
    "type": "div",
  },
  Object {
    "name": "Flex",
    "props": Object {},
    "style": Object {
      "display": "flex",
    },
    "system": Array [
      "alignItems",
      "justifyContent",
      "flexWrap",
      "flexDirection",
    ],
    "type": "Box",
  },
  Object {
    "name": "Grid",
    "props": Object {},
    "style": [Function],
    "system": Array [
      "alignItems",
      "justifyContent",
      "flexDirection",
    ],
    "type": "div",
  },
  Object {
    "name": "Text",
    "props": Object {},
    "style": Object {},
    "system": Array [
      "textAlign",
      "fontWeight",
    ],
    "type": "div",
  },
  Object {
    "name": "Heading",
    "props": Object {
      "m": 0,
    },
    "style": Object {
      "lineHeight": 1.25,
    },
    "system": Array [
      "textAlign",
      "fontWeight",
    ],
    "type": "h2",
  },
  Object {
    "name": "Link",
    "props": Object {
      "color": "blue",
    },
    "style": Object {},
    "system": Array [],
    "type": "a",
  },
  Object {
    "name": "Image",
    "props": Object {},
    "style": Object {
      "height": "auto",
      "maxWidth": "100%",
    },
    "system": Array [],
    "type": "img",
  },
  Object {
    "name": "Font",
    "props": Object {},
    "style": [Function],
    "system": Array [],
    "type": "div",
  },
  Object {
    "name": "DefaultLayout",
    "props": Object {
      "mx": "auto",
      "pb": 4,
      "pt": 3,
      "px": 3,
    },
    "style": Object {
      "maxWidth": "768px",
    },
    "system": Array [],
    "type": "div",
  },
]
`;

exports[`snapshot generates markdown components 1`] = `
Object {
  "Blockquote": [Function],
  "Code": [Function],
  "Divider": [Function],
  "Heading": [Function],
  "Pre": [Function],
  "Subhead": [Function],
  "Table": [Function],
  "Title": [Function],
}
`;

exports[`snapshot getData reads theme.json, lab.json, .jsx, and .md files 1`] = `
Object {
  "dirname": "/Users/himepro/DevelopProjects/gen/examples",
  "lab": Object {},
  "pages": Array [
    Object {
      "content": "<Box bg='tomato' color='white' p={4}>
  <h1>Hello Gen</h1>
</Box>
",
      "data": Object {
        "description": "Just a simple example file for gen",
        "title": "Hello",
      },
      "ext": ".jsx",
      "filename": "hello.jsx",
      "name": "hello",
      "raw": "---
title: Hello
description: Just a simple example file for gen
---
<Box bg='tomato' color='white' p={4}>
  <h1>Hello Gen</h1>
</Box>
",
    },
    Object {
      "content": "<Box bg='black' p={4}>
  <Box bg='white' py={4}>
    <Box mx='auto' px={3} style={{ maxWidth: 768 }}>
      <h1>Custom Layout</h1>
      {props.children}
    </Box>
  </Box>
</Box>
",
      "data": Object {},
      "ext": ".jsx",
      "filename": "layout.jsx",
      "name": "layout",
      "raw": "---
---
<Box bg='black' p={4}>
  <Box bg='white' py={4}>
    <Box mx='auto' px={3} style={{ maxWidth: 768 }}>
      <h1>Custom Layout</h1>
      {props.children}
    </Box>
  </Box>
</Box>
",
    },
    Object {
      "content": "
# No Layout (h1)

This markdown file doesn't specify a layout, so it gets a simple default one.

## H2

Hello there little markdown file.

## H3

How are you doing this fine evening

- [Hello](hello)
- [readme](readme)

---

[Made by Compositor](https://compositor.io/)
|
MIT License
",
      "data": Object {
        "title": "Home page",
        "twitter": Object {
          "card": "summary",
          "description": "Compositor JSX static site generator",
          "image": "kitten.png",
          "site": "@getcompositor",
          "title": "Compositor Gen",
        },
      },
      "ext": ".md",
      "filename": "index.md",
      "name": "index",
      "raw": "---
title: Home page
twitter:
  card: summary
  site: '@getcompositor'
  image: 'kitten.png'
  title: Compositor Gen
  description: Compositor JSX static site generator
---

# No Layout (h1)

This markdown file doesn't specify a layout, so it gets a simple default one.

## H2

Hello there little markdown file.

## H3

How are you doing this fine evening

- [Hello](hello)
- [readme](readme)

---

[Made by Compositor](https://compositor.io/)
|
MIT License
",
    },
    Object {
      "content": "
# Hello Markdown

This should render with Lab primitives

- [Link](https://compositor.io)

\`code\`

\`\`\`
preformatted text
\`\`\`
",
      "data": Object {
        "layout": "layout",
        "title": "Hello Markdown",
      },
      "ext": ".md",
      "filename": "readme.md",
      "layoutJSX": "<Box bg='black' p={4}>
  <Box bg='white' py={4}>
    <Box mx='auto' px={3} style={{ maxWidth: 768 }}>
      <h1>Custom Layout</h1>
      {props.children}
    </Box>
  </Box>
</Box>
",
      "name": "readme",
      "raw": "---
title: Hello Markdown
layout: layout
---

# Hello Markdown

This should render with Lab primitives

- [Link](https://compositor.io)

\`code\`

\`\`\`
preformatted text
\`\`\`
",
    },
  ],
  "theme": Object {
    "fonts": Array [
      "'Roboto Mono', monospace",
    ],
    "lineHeight": 1.625,
    "space": Array [
      0,
      4,
      8,
      16,
      32,
      64,
      128,
    ],
    "textColor": "#033",
  },
}
`;

exports[`snapshot render creates pages 1`] = `
Array [
  Object {
    "content": "<Box bg='tomato' color='white' p={4}>
  <h1>Hello Gen</h1>
</Box>
",
    "data": Object {
      "description": "Just a simple example file for gen",
      "title": "Hello",
    },
    "ext": ".jsx",
    "filename": "hello.jsx",
    "html": "<!DOCTYPE html><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1\\"><meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\"><meta name=\\"generator\\" content=\\"Compositor Gen\\"><title>Hello</title><meta name='description' content='Just a simple example file for gen'><link rel='stylesheet' href='https://fonts.googleapis.com/css?family=Roboto+Mono'><style>*{box-sizing:border-box}body{margin:0}</style><style data-styled-components=\\"cpFWnQ dHmUli\\">
/* sc-component-id: sc-bdVaJa */
.sc-bdVaJa {} .dHmUli{padding:32px;color:white;background-color:tomato;}
/* sc-component-id: sc-gzVnrw */
.sc-gzVnrw {} .cpFWnQ{font-family:'Roboto Mono',monospace;line-height:1.625;color:#033;}</style><body><div class=\\"sc-gzVnrw cpFWnQ\\"><div class=\\"sc-bdVaJa dHmUli\\" color=\\"white\\"><h1>Hello Gen</h1></div></div></body>",
    "name": "hello",
    "raw": "---
title: Hello
description: Just a simple example file for gen
---
<Box bg='tomato' color='white' p={4}>
  <h1>Hello Gen</h1>
</Box>
",
  },
  Object {
    "content": "<Box bg='black' p={4}>
  <Box bg='white' py={4}>
    <Box mx='auto' px={3} style={{ maxWidth: 768 }}>
      <h1>Custom Layout</h1>
      {props.children}
    </Box>
  </Box>
</Box>
",
    "data": Object {},
    "ext": ".jsx",
    "filename": "layout.jsx",
    "html": "<!DOCTYPE html><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1\\"><meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\"><meta name=\\"generator\\" content=\\"Compositor Gen\\"><title>undefined</title><link rel='stylesheet' href='https://fonts.googleapis.com/css?family=Roboto+Mono'><style>*{box-sizing:border-box}body{margin:0}</style><style data-styled-components=\\"cpFWnQ dHmUli ibxcLF dnFEMg iZjiKO\\">
/* sc-component-id: sc-bdVaJa */
.sc-bdVaJa {} .dHmUli{padding:32px;color:white;background-color:tomato;}.ibxcLF{padding:32px;background-color:black;}.dnFEMg{padding-top:32px;padding-bottom:32px;background-color:white;}.iZjiKO{margin-left:auto;margin-right:auto;padding-left:16px;padding-right:16px;}
/* sc-component-id: sc-gzVnrw */
.sc-gzVnrw {} .cpFWnQ{font-family:'Roboto Mono',monospace;line-height:1.625;color:#033;}</style><body><div class=\\"sc-gzVnrw cpFWnQ\\"><div class=\\"sc-bdVaJa ibxcLF\\"><div class=\\"sc-bdVaJa dnFEMg\\"><div class=\\"sc-bdVaJa iZjiKO\\" style=\\"max-width:768px\\"><h1>Custom Layout</h1></div></div></div></div></body>",
    "name": "layout",
    "raw": "---
---
<Box bg='black' p={4}>
  <Box bg='white' py={4}>
    <Box mx='auto' px={3} style={{ maxWidth: 768 }}>
      <h1>Custom Layout</h1>
      {props.children}
    </Box>
  </Box>
</Box>
",
  },
  Object {
    "content": "
# No Layout (h1)

This markdown file doesn't specify a layout, so it gets a simple default one.

## H2

Hello there little markdown file.

## H3

How are you doing this fine evening

- [Hello](hello)
- [readme](readme)

---

[Made by Compositor](https://compositor.io/)
|
MIT License
",
    "data": Object {
      "title": "Home page",
      "twitter": Object {
        "card": "summary",
        "description": "Compositor JSX static site generator",
        "image": "kitten.png",
        "site": "@getcompositor",
        "title": "Compositor Gen",
      },
    },
    "ext": ".md",
    "filename": "index.md",
    "html": "<!DOCTYPE html><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1\\"><meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\"><meta name=\\"generator\\" content=\\"Compositor Gen\\"><title>Home page</title><meta name='twitter:card' content='summary'><meta name='twitter:site' content='@getcompositor'><meta name='twitter:image' content='kitten.png'><meta name='twitter:title' content='Compositor Gen'><meta name='twitter:description' content='Compositor JSX static site generator'><link rel='stylesheet' href='https://fonts.googleapis.com/css?family=Roboto+Mono'><style>*{box-sizing:border-box}body{margin:0}</style><style data-styled-components=\\"cpFWnQ dHmUli ibxcLF dnFEMg iZjiKO PuZIC bculFG havvyu bPlFyU cVwVcu ecPPBI\\">
/* sc-component-id: sc-bdVaJa */
.sc-bdVaJa {} .dHmUli{padding:32px;color:white;background-color:tomato;}.ibxcLF{padding:32px;background-color:black;}.dnFEMg{padding-top:32px;padding-bottom:32px;background-color:white;}.iZjiKO{margin-left:auto;margin-right:auto;padding-left:16px;padding-right:16px;}
/* sc-component-id: sc-bxivhb */
.sc-bxivhb {} .havvyu{margin-bottom:16px;margin-top:0px;}
/* sc-component-id: sc-ifAKCX */
.sc-ifAKCX {} .bPlFyU{line-height:1.25;margin:0px;margin-bottom:16px;margin-top:32px;}
/* sc-component-id: sc-EHOje */
.sc-EHOje {} .cVwVcu{color:blue;}
/* sc-component-id: sc-gzVnrw */
.sc-gzVnrw {} .cpFWnQ{font-family:'Roboto Mono',monospace;line-height:1.625;color:#033;}
/* sc-component-id: sc-htoDjs */
.sc-htoDjs {} .PuZIC{max-width:768px;margin-left:auto;margin-right:auto;padding-bottom:32px;padding-top:16px;padding-left:16px;padding-right:16px;}
/* sc-component-id: sc-iwsKbI */
.sc-iwsKbI {} .bculFG{margin-bottom:16px;margin-top:32px;}
/* sc-component-id: sc-jzJRlG */
.sc-jzJRlG {} .ecPPBI{border:0;margin-top:16px;margin-bottom:16px;border-color:rgba(0,0,0,.125);}</style><style data-styled-components=\\"jqvTDw ipLFAD\\">
/* sc-component-id: sc-kAzzGY */
.sc-kAzzGY {} .jqvTDw{margin-bottom:16px;margin-top:32px;}
/* sc-component-id: sc-ckVGcZ */
.sc-ckVGcZ {} .ipLFAD{border:0;margin-top:16px;margin-bottom:16px;border-color:rgba(0,0,0,.125);}</style><body><div class=\\"sc-gzVnrw cpFWnQ\\"><div class=\\"sc-htoDjs PuZIC\\"><div><h1 class=\\"sc-iwsKbI bculFG\\" id=\\"user-content-no-layout-h1\\"><a href=\\"#user-content-no-layout-h1\\" style=\\"color:inherit;text-decoration:none\\">No Layout (h1)</a></h1>
<div class=\\"sc-bxivhb havvyu\\">This markdown file doesn&#x27;t specify a layout, so it gets a simple default one.</div>
<h2 class=\\"sc-ifAKCX bPlFyU\\" id=\\"user-content-h2\\"><a href=\\"#user-content-h2\\" style=\\"color:inherit;text-decoration:none\\">H2</a></h2>
<div class=\\"sc-bxivhb havvyu\\">Hello there little markdown file.</div>
<h2 class=\\"sc-ifAKCX bPlFyU\\" id=\\"user-content-h3\\"><a href=\\"#user-content-h3\\" style=\\"color:inherit;text-decoration:none\\">H3</a></h2>
<div class=\\"sc-bxivhb havvyu\\">How are you doing this fine evening</div>
<ul>
<li><a class=\\"sc-EHOje cVwVcu\\" href=\\"hello\\" color=\\"blue\\">Hello</a></li>
<li><a class=\\"sc-EHOje cVwVcu\\" href=\\"readme\\" color=\\"blue\\">readme</a></li>
</ul>
<hr class=\\"sc-jzJRlG ecPPBI\\"/>
<div class=\\"sc-bxivhb havvyu\\"><a class=\\"sc-EHOje cVwVcu\\" href=\\"https://compositor.io/\\" color=\\"blue\\">Made by Compositor</a>
|
MIT License</div></div></div></div></body>",
    "name": "index",
    "raw": "---
title: Home page
twitter:
  card: summary
  site: '@getcompositor'
  image: 'kitten.png'
  title: Compositor Gen
  description: Compositor JSX static site generator
---

# No Layout (h1)

This markdown file doesn't specify a layout, so it gets a simple default one.

## H2

Hello there little markdown file.

## H3

How are you doing this fine evening

- [Hello](hello)
- [readme](readme)

---

[Made by Compositor](https://compositor.io/)
|
MIT License
",
  },
  Object {
    "content": "
# Hello Markdown

This should render with Lab primitives

- [Link](https://compositor.io)

\`code\`

\`\`\`
preformatted text
\`\`\`
",
    "data": Object {
      "layout": "layout",
      "title": "Hello Markdown",
    },
    "ext": ".md",
    "filename": "readme.md",
    "html": "<!DOCTYPE html><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1\\"><meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\"><meta name=\\"generator\\" content=\\"Compositor Gen\\"><title>Hello Markdown</title><link rel='stylesheet' href='https://fonts.googleapis.com/css?family=Roboto+Mono'><style>*{box-sizing:border-box}body{margin:0}</style><style data-styled-components=\\"cpFWnQ dHmUli ibxcLF dnFEMg iZjiKO PuZIC bculFG havvyu bPlFyU cVwVcu ecPPBI bWArIO bESZpu kCSgPf\\">
/* sc-component-id: sc-bdVaJa */
.sc-bdVaJa {} .dHmUli{padding:32px;color:white;background-color:tomato;}.ibxcLF{padding:32px;background-color:black;}.dnFEMg{padding-top:32px;padding-bottom:32px;background-color:white;}.iZjiKO{margin-left:auto;margin-right:auto;padding-left:16px;padding-right:16px;}
/* sc-component-id: sc-bxivhb */
.sc-bxivhb {} .havvyu{margin-bottom:16px;margin-top:0px;}
/* sc-component-id: sc-ifAKCX */
.sc-ifAKCX {} .bPlFyU{line-height:1.25;margin:0px;margin-bottom:16px;margin-top:32px;}
/* sc-component-id: sc-EHOje */
.sc-EHOje {} .cVwVcu{color:blue;}
/* sc-component-id: sc-gzVnrw */
.sc-gzVnrw {} .cpFWnQ{font-family:'Roboto Mono',monospace;line-height:1.625;color:#033;}
/* sc-component-id: sc-htoDjs */
.sc-htoDjs {} .PuZIC{max-width:768px;margin-left:auto;margin-right:auto;padding-bottom:32px;padding-top:16px;padding-left:16px;padding-right:16px;}
/* sc-component-id: sc-iwsKbI */
.sc-iwsKbI {} .bculFG{margin-bottom:16px;margin-top:32px;}
/* sc-component-id: sc-jzJRlG */
.sc-jzJRlG {} .ecPPBI{border:0;margin-top:16px;margin-bottom:16px;border-color:rgba(0,0,0,.125);}
/* sc-component-id: sc-eNQAEJ */
.sc-eNQAEJ {} .bWArIO{margin-bottom:16px;margin-top:32px;}
/* sc-component-id: sc-kkGfuU */
.sc-kkGfuU {} .kCSgPf{font-family:Menlo,monospace;overflow:auto;margin-bottom:16px;margin-top:16px;padding:0px;padding-left:16px;font-size:14px;border-color:rgba(0,0,0,.125);}
/* sc-component-id: sc-iAyFgw */
.sc-iAyFgw {} .bESZpu{font-family:Menlo,monospace;font-size:14px;}</style><style data-styled-components=\\"irHpdu gvfqnR blNLxu\\">
/* sc-component-id: sc-jWBwVP */
.sc-jWBwVP {} .irHpdu{margin-bottom:16px;margin-top:32px;}
/* sc-component-id: sc-jAaTju */
.sc-jAaTju {} .blNLxu{font-family:Menlo,monospace;overflow:auto;margin-bottom:16px;margin-top:16px;padding:0px;padding-left:16px;font-size:14px;border-color:rgba(0,0,0,.125);}
/* sc-component-id: sc-jDwBTQ */
.sc-jDwBTQ {} .gvfqnR{font-family:Menlo,monospace;font-size:14px;}</style><body><div class=\\"sc-gzVnrw cpFWnQ\\"><div class=\\"sc-bdVaJa ibxcLF\\"><div class=\\"sc-bdVaJa dnFEMg\\"><div class=\\"sc-bdVaJa iZjiKO\\" style=\\"max-width:768px\\"><h1>Custom Layout</h1><div><h1 class=\\"sc-eNQAEJ bWArIO\\" id=\\"user-content-hello-markdown\\"><a href=\\"#user-content-hello-markdown\\" style=\\"color:inherit;text-decoration:none\\">Hello Markdown</a></h1>
<div class=\\"sc-bxivhb havvyu\\">This should render with Lab primitives</div>
<ul>
<li><a class=\\"sc-EHOje cVwVcu\\" href=\\"https://compositor.io\\" color=\\"blue\\">Link</a></li>
</ul>
<div class=\\"sc-bxivhb havvyu\\"><code class=\\"sc-iAyFgw bESZpu\\" font-size=\\"14\\">code</code></div>
<pre class=\\"sc-kkGfuU kCSgPf\\" font-size=\\"14\\"><code class=\\"sc-iAyFgw bESZpu\\" font-size=\\"14\\">preformatted text
</code></pre></div></div></div></div></div></body>",
    "layoutJSX": "<Box bg='black' p={4}>
  <Box bg='white' py={4}>
    <Box mx='auto' px={3} style={{ maxWidth: 768 }}>
      <h1>Custom Layout</h1>
      {props.children}
    </Box>
  </Box>
</Box>
",
    "name": "readme",
    "raw": "---
title: Hello Markdown
layout: layout
---

# Hello Markdown

This should render with Lab primitives

- [Link](https://compositor.io)

\`code\`

\`\`\`
preformatted text
\`\`\`
",
  },
]
`;
